crates/nix_health/src/check/direnv.rs

```diff
@@ -93,71 +93,74 @@ fn activation_check(local_flake: &std::path::Path, required: bool) -> Check {
     }
 }
 
-/// Information about a direnv install
-#[derive(Debug, Default, Serialize, Deserialize, Clone)]
-pub struct DirenvInstall {
-    /// Path to the direnv binary
-    pub bin_path: PathBuf,
+/// [Check] if direnv was already allowed in [project_dir]
 
-    /// Contents of `direnvrc`
-    pub direnv_config: Option<String>,
+fn is_direnv_allowed_on(project_dir: &std::path::Path) -> anyhow::Result<bool> {
+    let output = std::process::Command::new("direnv")
+        .args(["status", "--json"])
+        .current_dir(project_dir)
+        .output()?;
+    if output.status.success() {
+        let out = String::from_utf8_lossy(&output.stdout);
+        let status = DirenvStatus::from_json(&out)?;
+        // rc.allowed == 0 means that the .envrc is allowed, see: https://github.com/direnv/direnv/pull/1142#issuecomment-1848677532
+        Ok(status.state.found_rc.map_or(false, |rc| rc.allowed == 0))
```

Comment:
You can move this to a method on the struct, viz.:

```rust
impl DirenvStatus {
  fn is_allowed(self) -> bool { ... }
}
```

This way the caller doesn't have to care about the implementation; only the semantics.

Reply:
Actually it should be on `DirenvState`